sklearn/linear_model/logistic.py

```diff
@@ -1203,7 +1205,12 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("Tolerance for stopping criteria must be "
                              "positive; got (tol=%r)" % self.tol)
 
-        X, y = check_X_y(X, y, accept_sparse='csr', dtype=np.float64,
+        if self.solver in ['newton-cg']:
+            _dtype = [np.float64, np.float32]
```

Comment:
The idea is that previously `check_X_y` was converting  `X` and `y` into `np.float64`. This is fine, if the user passes a `list` as `X`, but if a user passes a `np.float32` willingly converting it to  `np.float64` penalizes them in memory and speed.

Therefore, we are trying to keep the data in `np.float32` if the user provides the data in such type.